ecosystem_cicd_tools/new_cicd/actions.py

```diff
@@ -45,6 +47,56 @@ def upload_assets_to_release(assets, release_name, repository, **_):
             os.environ.get('CIRCLE_USERNAME', 'earthmant')
         )
 
+    # Time to wait for the plugin to load
+    # And checking that everything was updated correctly
+    max_time = 180  # it should not take longer than 180 seconds.
+    min_time = 20  # It should definitely take longer than 20 seconds.
+    interval = 10  # We check every 10 seconds.
+    current = 0
+    while True:
+        if current < min_time:
+            current += interval
+            continue
+        if max_time <= current:
+            raise RuntimeError(
+                'Timed out waiting for marketplace plugin update.')
+        if checking_the_upload_of_the_plugin(repository, release_name, assets):
+            logging.logger.info(
+                'Verified plugin release in {} seconds'.format(current))
+            break
+        sleep(interval)
+        current += interval
+
+
+def checking_the_upload_of_the_plugin(repository, release_name, asset_workspace):
+    asset_workspace = list(asset_workspace.keys())
+
+    # github
+    latest_release = github.get_latest_release(repository)
+    assets_list_github = []
+    for asset in latest_release.get_assets():
+        assets_list_github.append(asset.label)
+
+    # marketplace
+    assets_list_marketplace = marketplace.get_assets(repository)
+    node_types = marketplace.get_node_types_for_plugin_version(repository.name,
+                                                               release_name)
+    if not node_types:
+        raise RuntimeError(
+            'Failed to update marketplace with plugin release.')
+
+    # s3
+    assets_list_s3 = s3.get_assets(repository.name, release_name)
+
+    for asset in asset_workspace:
+        if "wgn.md5" not in asset and asset not in assets_list_marketplace \
+                or asset not in assets_list_github \
+                or asset not in assets_list_s3:
+            raise RuntimeError(
+                'Failed to update marketplace with plugin release.')
```

Comment:
I think that this should be rewritten so that when we raise RuntimeError, we say specifically which asset was not found in which list. Consider that sometime this error might happen and we will want to know exactly which assets and which list is the problem. You can do it like this:

```python
def example(list_1, list_2, list_3, assets):
   problems = []
   for asset in assets:
       if asset.endswith('wgn.md5'):
           continue
       if asset not in list_1:
           problems.append('{} not found in list_1'.format(asset))
       if asset not in list_2:
           problems.append('{} not found in list_2'.format(asset))
       if asset not in list_2:
           problems.append('{} not found in list_3'.format(asset))
  if problems:
       raise RuntimeError('Failed to verify all assets: {}'.format(problems))
```